**Girish Palya**

```
The diagram shows the map of city bikes movement from district to district - the probability of bicycle relocation from one to another district in one day. At the beginning of the season, all 2000 bicycles will be left in B district.  Show the behaviour of the system in time. What is the steady state? Is it a good idea to leave all bicycles in B district? Where the provider should build new docking stations when the number of bicycles will be growing.  
New version: In which Warsaw district we can found most of the electric scooters? [ we have no docking station]
```

In [31]:
import numpy as np
markov_m = [[0.4, 0.2, 0.4],
           [0.6, 0.3, 0.1],
           [0.1, 0.4, 0.5]] 
markov = np.matrix(markov_m)

distrib_m = [[0, 2000, 0]]
distrib = np.matrix(distrib_m)

# Iterate over many time periods
for t in range(10):
    print(f'After {t} time period, allocation A:B:C = {distrib[0,0]:.0f}:{distrib[0,1]:.0f}:{distrib[0,2]:.0f}')
    distrib = distrib * markov

After 0 time period, allocation A:B:C = 0:2000:0
After 1 time period, allocation A:B:C = 1200:600:200
After 2 time period, allocation A:B:C = 860:500:640
After 3 time period, allocation A:B:C = 708:578:714
After 4 time period, allocation A:B:C = 701:601:698
After 5 time period, allocation A:B:C = 711:600:690
After 6 time period, allocation A:B:C = 713:598:689
After 7 time period, allocation A:B:C = 713:598:690
After 8 time period, allocation A:B:C = 713:598:690
After 9 time period, allocation A:B:C = 713:598:690


**Comments**
- Steady state after 7 time periods (713 : 598 : 690)
- Not a good idea to leave all bycycles in B
- Start with 713:598:690 distribution (A:B:C)
- New docking stations should be built in the same proportion as steady state (more in A)
- More electric scooters in Centrum(?)